src/client/Client.cc

```diff
@@ -12612,6 +12841,22 @@ int Client::_rename(Inode *fromdir, const char *fromname, Inode *todir, const ch
     if (fromdir_root != todir_root) {
       return -EXDEV;
     }
+
+    fromdir_root =
+      fromdir->quota.is_user_enable()? fromdir : get_user_quota_root(fromdir);
+    todir_root =
+      todir->quota.is_user_enable()? todir : get_user_quota_root(todir);
+    if (fromdir_root != todir_root) {
+      return -EXDEV;
+    }
+
+    fromdir_root =
+      fromdir->quota.is_group_enable()? fromdir : get_group_quota_root(fromdir);
+    todir_root =
+      todir->quota.is_group_enable()? todir : get_group_quota_root(todir);
+    if (fromdir_root != todir_root) {
+      return -EXDEV;
+    }
```

Comment:
why not make quota.is_enable() cover user/group quota?

Reply:
I have added quota.is_any_enable()  to cover directory/user/group quota, But there are scenarios where directories, users, and groups quotas need to be judged separately, So these independent judgement functions are preserved. Thanks.